### PR TITLE
Fix hard coded paths in test_deploy_subcommand

### PR DESCRIPTION
**Pull request by aptxkid**
* fixed hard coded paths in test_deploy_subcommand
* fixed hard coded os.sep length in deploy_subcommand
* consolidated the tests by using genty

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @aptxkid has signed the CLA. Thanks for the pull request!

File: test/unit/subcommands/test_deploy_subcommand.py [hunk 1]
```diff
@@ -34,56 +35,33 @@ def test_deploy_binaries_and_conf_deploys_both_conf_and_binary_for_remote_host(s
...
-        deploy_subcommand = DeploySubcommand()
-        deploy_subcommand._deploy_binaries_and_conf(
-            'localhost',
-            'username',
-            'exec',
-            '/path/to/exec',
-            '/home/.clusterrunner/clusterrunner_prime.conf'
-        )
-        self.assertTrue(mock_DeployTarget_instance.deploy_conf.called)
-
```
Comment (tjlee0909) [inline test/unit/subcommands/test_deploy_subcommand.py]: What was wrong with this test that caused you to delete it?

Comment (aptxkid) [inline test/unit/subcommands/test_deploy_subcommand.py]: test_deploy_binaries_and_conf_doesnt_deploy_conf_if_localhost_with_same_in_use_conf
test_deploy_binaries_and_conf_deploys_conf_if_localhost_with_diff_in_use_conf
test_deploy_binaries_and_conf_deploys_binaries_if_localhost_and_different_executable_path_in_use
test_deploy_binaries_and_conf_doesnt_deploy_binaries_if_localhost_and_same_executable_path_in_use

are not very DRY. So I removed two tests and added asserts for deploy_conf for the remaining two. Right now it's testing (conf_in_use, exe_in_use) and (conf_not_in_use, exe_not_in_use) in two unit tests. I realize that it's missing the coverage of (conf_in_use, exe_not_in_use) and (conf_not_in_use, exe_in_use). Actually, I am going to try to consolidate these four unit tests into one.

**Comment by aptxkid** [comment]
@tjlee0909 I've updated the tests using genty.

**Comment by tjlee0909** [comment]
:+1:

**Pull request merged**